starter-code/iteration-3/main.ts

```diff
@@ -1,8 +1,46 @@
 // Add the reference to the interface
+import {TodoItemInterface, TodoListInterface} from "./interfaces"
 
 // Create class TodoItem that implements the corresponding interface
+class TodoItem implements TodoItemInterface {
+  constructor(public title: string) {}
+  updatedAt: Date = new Date();
+  status: boolean = true;
+  toggleStatus() {
+    if (this.status) {
+      this.status = false;
+    } else {
+    this.status = true;
+    }
+  }
+}
 
 // Create class TodoList that implements the corresponding interface
+class TodoList implements TodoListInterface {
+  todoList: Array<TodoItem> = [];
+  addTask(item:TodoItem): number{
+    this.todoList.push(item)
+    return this.todoList.length;
+  };
+  listAllTasks() {
+    for (var i = 0; i < this.todoList.length; i++) {
+      console.log(this.todoList[i]);
+    }
+  }
+  deleteTask(item:TodoItem): number {
+    this.todoList.splice(this.todoList.indexOf(item), 1);
+    console.log("Item deleted!");
+    return this.todoList.length;
+  }
+  listUncomplete() {
+    for (var i = 0; i < this.todoList.length; i++) {
```

Comment:
you could also improve this function like this:
`this.tasks.forEach((task) => {
      if (!task.status) {
        console.log(`Pending: ${task.title}, ${task.updatedAt}`);
      }
    });`